### DISCUSSION TITLE
What is the difference between the data obtained by async-profiler and data captured by eBPF agents?

### DISCUSSION THREAD

**Discussion by Loveen01** [async-profiler/async-profiler#771] [Q&A] [answered]
Hello does the async-profiler capture exactly the same data/stack traces that is brought from typical eBPF agents/profilers? They seem to both use perf_event_open, except that async-profiler is non-ebpf based. But are the resulting frames on the flamegraphs similar, or does the async-profiler get less information from kernel stack than eBPF agents ? Really interested to know. Thanks alot in advance.

**Comment by apangin** [comment, marked as answer]
async-profiler gets similar kernel stack traces as `perf` or bpf based tools.
Make sure to use `-e cycles` instead of `-e cpu` for the most accurate kernel stack traces (this may not be available on some virtualized systems).

Unlike system-wide profiling agents, async-profiler works with one process at a time. On the other hand, system-wide profilers cannot get as complete and accurate Java stack traces as async-profiler does.
